### DISCUSSION TITLE
Help on architecture when using dotnet as backend

### DISCUSSION THREAD

**Discussion by jvskriubakken** [nrwl/nx#13753] [Show and Tell]
Hi,

I am working in a company where we implement the backend using C# and the dotnet platform. And we want to setup a mono-repo with several applications with frontends implemented using React and Typescript.

We are trying to figure out the idiomatic setup.  We want to have an "integreated repo" for getting the most out of Nx.

I have a few questions which I hope some of you might help with answering.

We are studying the nx-dotnet plugin, and if we following the getting started guide we end up with creating an nx app that has both frontend (react) and backend in the same nx project:
`npx nx g @nx-dotnet/core:app my-api --test-template nunit --language C#`
...as we choose the react template from dotnet in one of the following steps.

Is this recommended? I believe that preferred way with nx is to have the frontend app and backend webapp as two separate nx projects. Otherwise Nx will not be responsible for the frontend "compiling" to js, which I believe is not good. As we want to use Nx´s expertise on that ;)

Thanks!

**Comment by rfprod** [comment]
From a build and test standpoint, I would recommend separating concerns, no matter which front-end framework you end up using.

**Reply by jvskriubakken** [reply]
Thanks! Yes, that sounds the best! I got a long and good answer from Craigory Coppola on Nx Slack support channel, which I can supplement on this discussion later.

**Reply by ybybybybyb** [reply]
@jvskriubakken What's your final setup a year later?

We have a project with 2 dotnet webapi backends and 2 Nuxt3(VueJS) frontend and are observing using NX to manage the lifecycle of the projects.
